src/core/pointcloud/qgspointcloudrequest.h

```diff
@@ -55,18 +56,18 @@ class CORE_EXPORT QgsPointCloudRequest
 
     /**
      * Returns the rectangle from which points will be taken, in point cloud's crs. If the returned rectangle is empty, then no filter rectangle is set.
-     * \since QGIS 3.30
+     * \since QGIS 3.40
      */
-    QgsRectangle filterRect() const { return mFilterRect; }
+    QgsGeometry filterGeometry() const { return mFilterGeometry; }
 
     /**
      * Sets the rectangle from which points will be taken, in point cloud's crs. An empty rectangle removes the filter.
-     * \since QGIS 3.30
+     * \since QGIS 3.40
      */
-    void setFilterRect( QgsRectangle extent ) { mFilterRect = extent; }
+    void setFilterGeometry( const QgsGeometry &geometry ) { mFilterGeometry = geometry; }
```

Comment:
I'm -1 to this change. An intersection test against a polygon is ALWAYS going to be considerably more expensive than an intersection test against a rectangle, so we're incurring a considerable speed regression for all point cloud use for just this one rare use case.

There's two options I see here:

1. Add setFilterGeometry as a new function in ADDITION to the existing setFilterRect function, and leave the optimised rectangle checks in place in addition to the new logic.
2. Stick with filter rect checks in the point cloud request, and then from the 3d scene just pass the extent of the rotated geometry as the filter rect. Add the extra filtering in the 3d scene code only, using prepared GEOS geometry intersects for performance. (And only do that when the rotation is non zero, so we don't incur any cost for the default rotation=0 setting).

Option 1 would be more flexible, but is considerably trickier to get right. You'd need to ensure that the geometry intersection tests in the point cloud requests ALWAYS use a prepared GEOS geometry, and that the rectangular bounding box of the geometry is still used when traversing the point cloud index. You'd also need to handle the situation when both filter rect + filter geometry are set elegantly.

Reply:
Thanks for the advice. Option 2 is now implemented.